src/components/Button/Button.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ComponentPropsWithoutRef } from "react";
+
+export interface ButtonProps extends ComponentPropsWithoutRef<"button"> {
+  children?: React.ReactNode;
+  preIcon?: JSX.Element;
+  posIcon?: JSX.Element;
+  variant: string;
+}
+
+export function Button(props: ButtonProps) {
```

Comment:
You should have a way to store state of your button like a 
useState.
store value of button on click 

Because the Pressed version is different from Default

Reply:
Would you like me to store the state of the button (pressed, hover, disabled, default) to apply the different Tailwaind classes?

Because I use the variant here to apply the different states depending on the type of button.

Reply:
Nope what i was meaning is to display "pressed" button you have to store a isActive state . 

It's internal of the button you have to manage the actual state is active or not directly in your button in tsx. 
otherwise you can't reproduce the pressed state (Pressed state is when you have click on button you freese the button in this state, if pressed is not this this comment is not usefull)
I have made it in my pr have look on this https://github.com/massalabs/ui-kit/pull/36